app/controllers/welcome_controller.rb

```diff
@@ -1,5 +1,10 @@
 class WelcomeController < ApplicationController
   def index
+    if current_user
```

Comment:
Okay, I guess that what would be better would be to make user#show the root, and redirect to the landing page if the user is not logged in, rather than making welcome#index the root and redirecting to user#show. I'll make that change.